python/MooseDocs/common/get_content.py

```diff
@@ -74,8 +74,12 @@ def _doc_import(root_dir, content=None):
     """
     # Check root_dir
     if not os.path.isdir(root_dir):
-        LOG.error('The "root_dir" must be a valid directory: %s', root_dir)
-        return None
+        if not optional:
+            LOG.error('The "root_dir" must be a valid directory: %s', root_dir)
+        else:
+            LOG.warning("Part of the content specified in 'config.yml' is not included in documentation since " +
+                        root_dir + ' is not a valid directory')
+        return set()
```

Comment:
```suggestion
        return None
```

Let's continue to return `None`, rather than an empty set. Judging from the passing tests we don't do error checking against this return (or an empty set is sufficient), but this is consistent with the other error conditions at this level of the method.

Reply:
I really need a valid return to avoid changing the for loop afterwards

the error conditions are throwing so I dont think the for loop matters. This is moving along instead

Reply:
umm lemme upload the changes I would have to make and you decide

Reply:
done